src/vs/platform/environment/node/userDataPath.js

```diff
@@ -60,6 +60,11 @@
 				return path.join(appDataPath, productName);
 			}
 
+			// 3. Support debugging mac OSS CLI
+			if (process.platform === 'darwin' && cliArgs['exec-path']) {
+				return cliArgs['exec-path'];
```

Comment:
I do not really understand this change. The method is for resolving the user-data-path, how does this need to be the `exec-path`?

Reply:
This change allows for the correct telemetry file to be pulled up when running with the --exec-path argument. It only affects running OSS from terminal with the following command: `vscode/.build/electron/Code\ -\ OSS.app/Contents/MacOS/Electron vscode/out/cli.js --exec-path /Applications/Visual\ Studio\ Code.app --telemetry`.
I can take it out, considering that in the shipped product, `code --telemetry` would work without this change.

Reply:
Will undo this change
See https://github.com/microsoft/vscode/pull/131213#issuecomment-904184370